periphery/src/managers/UniswapNFTManager.sol

```diff
@@ -49,9 +46,12 @@ contract UniswapNFTManager is IManager {
             ERC20 token0 = borrower.TOKEN0();
             ERC20 token1 = borrower.TOKEN1();
 
-            (uint256 burned0, uint256 burned1, , ) = borrower.uniswapWithdraw(lower, upper, uint128(liquidity));
-            token0.safeTransferFrom(msg.sender, address(this), burned0);
-            token1.safeTransferFrom(msg.sender, address(this), burned1);
+            (uint256 burned0, uint256 burned1, uint256 collected0, uint256 collected1) = borrower.uniswapWithdraw(
+                lower,
+                upper,
+                uint128(liquidity),
+                address(this)
```

Comment:
By specifying `address(this)` as the recipient, we avoid having to do a separate `safeTransferFrom` from the borrower to `address(this)`. This also means we don't need to set the `allowances` array when calling modify, which saves gas in and of itself.